pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -2306,6 +2306,310 @@ def init_arc_param(arctype):
+                        * model.p_lambda_Pipeline[l, l_tilde]
+                        * model.p_eta_PipelineOperationsEmissionsCoefficient[a]
+                        for l in model.s_L
+                        if (l, l_tilde) in model.s_LLA
+                    )
+                    for l_tilde in model.s_L
+                )
+                for t in model.s_T
+            )
+            * model.model_units["time"]
+        )
+
+    model.e_TotalPipeOperationsEmissions = Expression(
+        model.s_A,
+        rule=TotalPipelineOperationsEmissionsRule,
+        doc="Total pipeline operations emissions",
+    )
+
+    def TotalPipelineInstallationEmissionsRule(model, a):
+        return sum(
+            sum(
+                sum(model.vb_y_Pipeline[l, l_tilde, d] for d in model.s_D)
+                * model.p_lambda_Pipeline[l, l_tilde]
+                * model.p_eta_PipelineInstallationEmissionsCoefficient[a]
+                for l in model.s_L
+                if (l, l_tilde) in model.s_LLA
+            )
+            for l_tilde in model.s_L
+        )
+
+    model.e_TotalPipeInstallEmissions = Expression(
+        model.s_A,
+        rule=TotalPipelineInstallationEmissionsRule,
+        doc="Total pipeline installation emissions",
+    )
+
+    def TotalDisposalEmissionsRule(model, a):
+        return model.v_F_TotalDisposed * model.p_eta_DisposalEmissionsCoefficient[a]
+
+    model.e_TotalDisposalEmissions = Expression(
+        model.s_A, rule=TotalDisposalEmissionsRule, doc="Total disposal emissions"
+    )
+
+    def TotalStorageEmissionsRule(model, a):
+        return (
+            sum(
+                sum(
+                    model.v_L_Storage[s, t] * model.p_eta_StorageEmissionsCoefficient[a]
+                    for s in model.s_S
+                )
+                for t in model.s_T
+            )
+            * model.model_units["time"]
+        )
+
+    model.e_TotalStorageEmissions = Expression(
+        model.s_A, rule=TotalStorageEmissionsRule, doc="Total storage emissions"
+    )
+
+    def TotalTreatmentEmissionsRule(model, a):
+        return sum(
+            sum(
+                sum(
+                    model.v_F_TreatmentFeedTech[r, wt, t]
+                    * model.p_eta_TreatmentEmissionsCoefficient[wt, a]
+                    for wt in model.s_WT
+                )
+                for r in model.s_R
+            )
+            for t in model.s_T
+        )
+
+    model.e_TotalTreatmentEmissions = Expression(
+        model.s_A, rule=TotalTreatmentEmissionsRule, doc="Total treatment emissions"
+    )
+
+    def TotalEmissionsByComponentRule(model, a):
+        return (
+            model.e_TotalTruckingEmissions[a]
+            + model.e_TotalPipeOperationsEmissions[a]
+            + model.e_TotalPipeInstallEmissions[a]
+            + model.e_TotalDisposalEmissions[a]
+            + model.e_TotalStorageEmissions[a]
+            + model.e_TotalTreatmentEmissions[a]
+        )
```

Comment:
Do we not include the air emissions by the subsurface activities?

Reply:
@JavalVyas2000 what subsurface activities? I'm not sure what you mean here.

Reply:
@tarnold17 I mean emissions happening during the fracking process.

Reply:
Got it - that is not something we are including right now. We could add it later if needed.